thirdTask/thirdTask/ViewController.swift

```diff
@@ -24,54 +24,83 @@ class ViewController: UIViewController, UITableViewDataSource, UITableViewDelega
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        adjustLayoutForOrientation()
+    }
     
-    // MARK: - UITableViewDataSource Methods
+    func setupConstraints() {
+        colorsTable.translatesAutoresizingMaskIntoConstraints = false
+        textDescription.translatesAutoresizingMaskIntoConstraints = false
+
+        let constraintsLandscape = [
+            colorsTable.topAnchor.constraint(equalTo: view.topAnchor),
+            colorsTable.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            colorsTable.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 0.4),
+            colorsTable.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            
+            textDescription.topAnchor.constraint(equalTo: view.topAnchor),
+            textDescription.leadingAnchor.constraint(equalTo: colorsTable.leadingAnchor),
+            textDescription.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+            textDescription.bottomAnchor.constraint(equalTo: view.bottomAnchor)
+            
+        ]
+        
+        self.landscapeConstraints = constraintsLandscape
+    }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    func adjustLayoutForOrientation() {
+        if UIDevice.current.orientation.isLandscape {
+            NSLayoutConstraint.activate(landscapeConstraints)
+            navigationTitle.title = ""
+            
+        } else {
+            NSLayoutConstraint.deactivate(landscapeConstraints)
+            navigationTitle.title = "Colors"
+            
+        }
+    }
+    
+    var landscapeConstraints:  [NSLayoutConstraint] = []
+    
+    func tableView(_ tableView:  UITableView,  numberOfRowsInSection section: Int) -> Int {
         return colors.count
     }
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "colorCell", for: indexPath)
+    func tableView(_ tableView:  UITableView,  cellForRowAt indexPath:  IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier:  colorCell, for:  indexPath)
         
-        // Get the color and name for the current row
         let (color, name) = colors[indexPath.row]
         
-        // Set the cell's background color
         cell.backgroundColor = color
         cell.textLabel?.text = name
         
-        // Set the text color based on the name
         if name == "Black" {
             cell.textLabel?.textColor = .white
-        }  
-        if name != "Black" {
+        } else {
             cell.textLabel?.textColor = .black
         }
         
         cell.selectionStyle = .none
-
+        
         return cell
     }
     
-    // Optional: Set the height for each row
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
```

Comment:
when creating a custom cell, you shouldn't specify the height for a table view cell. According to the content inside the cell and its constraints the height will be determined.